app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapRoutesRefresher.kt

```diff
@@ -0,0 +1,94 @@
+package org.p2p.wallet.swap.jupiter.statemanager
+
+import java.math.BigDecimal
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import org.p2p.core.token.Token
+import org.p2p.core.utils.fromLamports
+import org.p2p.core.utils.toLamports
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.swap.jupiter.repository.model.JupiterSwap
+import org.p2p.wallet.swap.jupiter.repository.model.JupiterSwapRoute
+import org.p2p.wallet.swap.jupiter.repository.routes.JupiterSwapRoutesRepository
+import org.p2p.wallet.swap.jupiter.repository.transaction.JupiterSwapTransactionRepository
+import org.p2p.wallet.utils.toBase58Instance
+
+class SwapRoutesRefresher(
```

Comment:
im not sure about this naming

this is class with reusable logic that can be triggered by different actions